controllers/crd/crd_controller.go

```diff
@@ -21,18 +21,21 @@ var (
 	log = logf.Log.WithName("controller_crd")
 )
 
-// RegisterReconciler creates a new Descheduler Reconciler and registers it into manager.
-func RegisterReconciler(mgr manager.Manager, restartCh chan<- struct{}) error {
-	return add(mgr, newReconciler(mgr, restartCh))
+// RegisterReconciler creates a new CRD Reconciler and registers it into manager.
+// nonCachedClient should be a direct client (pre-manager) to query CRDs at boot.
+func RegisterReconciler(ctx context.Context, nonCachedClient client.Client, mgr manager.Manager, restartCh chan<- struct{}) error {
```

Comment:
let's pass the boolean from main, instead of reading it again here


```suggestion
func RegisterReconciler(mgr manager.Manager, restartCh chan<- struct{}, persesAvailableOnBoot bool) error {
```